File: platform/spark-ingestion/packages/core/src/ingestion_models/requests.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Any, Dict, List, Optional


@dataclass
class CollectionJobRequest:
    runId: str
    endpointId: str
    sourceId: str
    endpointName: str
    connectionUrl: str
    schemas: List[str]
    projectId: Optional[str] = None
    labels: Optional[List[str]] = None
    config: Optional[Dict[str, Any]] = None


@dataclass
class CatalogRecordOutput:
    id: str
    projectId: Optional[str]
    domain: str
    labels: List[str]
    payload: Dict[str, Any]


@dataclass
class CollectionJobResult:
    recordsPath: Optional[str]
    recordCount: int
    logs: List[Dict[str, Any]]


@dataclass
class PreviewRequest:
    datasetId: str
    schema: str
    table: str
    endpointId: str
    unitId: str
    templateId: Optional[str] = None
    parameters: Dict[str, Any] = field(default_factory=dict)
    connectionUrl: Optional[str] = None
    limit: Optional[int] = 50


@dataclass
class IngestionUnitRequest:
    endpointId: str
    unitId: str
    sinkId: Optional[str] = None
    checkpoint: Optional[Dict[str, Any]] = None
    stagingProviderId: Optional[str] = None
    policy: Optional[Dict[str, Any]] = None
    mode: Optional[str] = None
    dataMode: Optional[str] = None
    sinkEndpointId: Optional[str] = None
    cdmModelId: Optional[str] = None
    filter: Optional[Dict[str, Any]] = None
    transientState: Optional[Dict[str, Any]] = None
    transientStateVersion: Optional[str] = None


@dataclass
class IngestionUnitResult:
    newCheckpoint: Optional[Dict[str, Any]]
    stats: Dict[str, Any]
    records: Optional[List[Dict[str, Any]]] = None
    transientState: Optional[Dict[str, Any]] = None
    stagingPath: Optional[str] = None
    stagingProviderId: Optional[str] = None


__all__ = [
    "CollectionJobRequest",
    "CatalogRecordOutput",
    "CollectionJobResult",
    "PreviewRequest",
    "IngestionUnitRequest",
    "IngestionUnitResult",
]

File: platform/ucl-core/internal/core/requests.go
package core

// =============================================================================
// INGESTION REQUEST/RESULT MODELS
// Request and response types for ingestion operations.
// =============================================================================

// CollectionJobRequest triggers a metadata collection run.
type CollectionJobRequest struct {
	RunID         string
	EndpointID    string
	SourceID      string
	EndpointName  string
	ConnectionURL string
	Schemas       []string
	ProjectID     string
	Labels        []string
	Config        map[string]any
}

// CollectionJobResult contains collection run outcome.
type CollectionJobResult struct {
	RecordsPath string
	RecordCount int
	Logs        []map[string]any
}

// CatalogRecordOutput represents a catalog record for output.
type CatalogRecordOutput struct {
	ID        string
	ProjectID string
	Domain    string
	Labels    []string
	Payload   map[string]any
}

// PreviewRequest requests a dataset preview.
type PreviewRequest struct {
	DatasetID     string
	Schema        string
	Table         string
	EndpointID    string
	UnitID        string
	TemplateID    string
	Parameters    map[string]any
	ConnectionURL string
	Limit         int
}

// IngestionUnitRequest triggers an ingestion unit run.
type IngestionUnitRequest struct {
	EndpointID            string
	UnitID                string
	SinkID                string
	Checkpoint            map[string]any
	StagingProviderID     string
	Policy                map[string]any
	Mode                  string
	DataMode              string
	SinkEndpointID        string
	CdmModelID            string
	Filter                map[string]any
	TransientState        map[string]any
	TransientStateVersion string
}

// IngestionUnitResult contains ingestion run outcome.
type IngestionUnitResult struct {
	NewCheckpoint     map[string]any
	Stats             map[string]any
	Records           []map[string]any
	TransientState    map[string]any
	StagingPath       string
	StagingProviderID string
	Staging           []map[string]any
	StageRef          string
	BatchRefs         []string
	BytesStaged       int64
	RecordsStaged     int64
}
